File: leetcode2/1easy/최원준/Q2614.java
package Leetcode.최원준;

/*
1. 아이디어 :


2. 시간복잡도 :
O( )

3. 자료구조/알고리즘 :

 */

public class Q2614 {
    class Solution {
        private boolean isPrime(int n) {
            if (n==1) return false;
            if (n<=3) return true;
            for (int i=2; i<=Math.sqrt(n); i++) {
                if (n%i==0) return false;
            }
            return true;
        }
        public int diagonalPrime(int[][] nums) {
            int n = nums.length;
            int ans = 0;

            for (int i=0; i<n; i++) {
                if (isPrime(nums[i][i])) ans = Math.max(ans, nums[i][i]);
                if (isPrime(nums[i][n-i-1])) ans = Math.max(ans, nums[i][n-i-1]);
            }
            return ans;
        }
    }
}

File: leetcode2/1easy/노형민/Q2614.cs
public class Solution
{
    public int DiagonalPrime(int[][] nums)
    {
        int maxPrime = 0;

        for (int i = 0; i < nums.Length; i++)
        {
            int diagonalValue1 = nums[i][i];
            int diagonalValue2 = nums[i][nums.Length - 1 - i];

            if (diagonalValue1 > maxPrime && IsPrime(diagonalValue1))
            {
                maxPrime = Math.Max(maxPrime, diagonalValue1);
            }

            if (diagonalValue2 > maxPrime && IsPrime(diagonalValue2))
            {
                maxPrime = Math.Max(maxPrime, diagonalValue2);
            }
        }

        return maxPrime;
    }

    private bool IsPrime(int num)
    {
        int sqrt = (int)Math.Sqrt(num);
        if (num == 1)
        {
            return false;
        }

        for (int i = 2; i <= sqrt; i++)
        {
            if (num % i == 0)
            {
                return false;
            }
        }

        return true;
    }
}